pkg/operator/customroute/custom_route_conditions.go

```diff
@@ -0,0 +1,205 @@
+package customroute
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	"github.com/openshift/library-go/pkg/route/routeapihelpers"
+	"github.com/openshift/library-go/pkg/route/routecomponenthelpers"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func ensureDefaultConditions(conditions []metav1.Condition) []metav1.Condition {
+	for _, conditionType := range []string{"Progressing", "Degraded"} {
+		condition := findCondition(conditions, conditionType)
+		if condition == nil {
+			conditions = append(conditions, metav1.Condition{
+				LastTransitionTime: metav1.Now(),
+				Type:               conditionType,
```

Comment:
this looks weird/incorrect.  Conditions on operator resources are suffixed with Progressing and Degraded, but they shouldn't actually *be* progressing or degraded.

Reply:
these won't be conditions in an operator resource but in the `ingress.config` object's status

Reply:
Just echoing, what @stlaz said. This is the `Ingress.Status.ComponentRoutes[].Conditions[]` from `github.com/openshift/api/config/v1/types_ingress.go`